test/suite/test_compact04.py

```diff
@@ -30,13 +30,25 @@
 #   Test the accuracy of compact work estimation.
 #
 
-import time, wiredtiger, wttest
+import wttest
 from wiredtiger import stat
+from wtscenario import make_scenarios
 
-# Test the accuracy of compact work estimation.
+# Test if compaction proceeds depending on the free_space_target field and the accuracy of compact
+# work estimation.
 class test_compact04(wttest.WiredTigerTestCase):
 
-    # Keep debug messages on by default. This is useful for diagnosing spurious test failures.
+    free_space_target = [
```

Comment:
Please separate your changes into their own test, so that this test can focus only on the accuracy of the compact work estimation.

Reply:
That's a good idea, reverted the changes in 363c9b7d6ec3141bcbe97e25057e535e54c1554e and added a new test in c3cd51497a3e6074ae49bdd7c76556f756441f0e.